### PR TITLE
Login user and respond with JSON along with auth cookie

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

**Comment by amitaibu** [comment]
Still needs tests.

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch any objection? :)

**Comment by amitaibu** [comment]
fyi, I've started working on [this](https://github.com/Gizra/angular-restful-auth) Angular example app (not making it too much public yet, as I want to style it + write a blog post about it)

And it's actually already working! :)

**Comment by e0ipso** [comment]
Count me in on that too.

**Comment by amitaibu** [comment]
> Count me in on that too.

You mean on the Angular app? 

Also, I believe this PR is ready to ship

**Comment by amitaibu** [comment]
Merged.

File: plugins/restful/user/login_cookie/1.0/login_cookie__1_0.inc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+$plugin = array(
+  'label' => t('Login'),
+  'description' => t('Login a user and return a JSON along with the authentication cookie..'),
```
Comment (e0ipso) [inline plugins/restful/user/login_cookie/1.0/login_cookie__1_0.inc:5]: Extra trailing '.'

**Comment by e0ipso** [comment]
Sorry for commenting late, but I did not have the time earlier.

I don't see any tests for `login_cookie`, is that intentional?

**Comment by e0ipso** [comment]
Also, maybe it deserves a little documentation on how to set the cookie jar to capture the returned cookie. Perhaps an example using curl.

**Comment by amitaibu** [comment]
> I don't see any tests for login_cookie, is that intentional?

No, just an oversight, I'll write ones soon.

> Perhaps an example using curl.

I actually didn't check with CURL, but it's a good point. Maybe also point in the readme to the Angular app.

**Comment by e0ipso** [comment]
I like curl because you can brainlessly c&p to your terminal, but the Angular app is a real world case scenario. Up to you :smiley:

**Pull request closed without merging**